1.1-category-the-essence-of-composition.md

```diff
@@ -0,0 +1,54 @@
+```Haskell
+f :: A -> B
+```
+```swift
+let f : (A) -> B
+```
+................
+```Haskell
+g :: B -> C
+```
+```swift
+let g: (B) -> C
+```
+...............
+```Haskell
+g . f
+```
+```swift
+compose(g, f)
+```
+..............
+```Haskell
+f :: A -> B
+g :: B -> C
+h :: C -> D
+h . (g . f) == (h . g) . f == h . g . f
+```
+```swift
+let f: (A) -> B
+let g: (B) -> C
+let h: (C) -> D
+
+compose(h, compose(g, f)) == compose(compose(h, g), f)
+```
+..............
+```Haskell
+id :: a -> a
+id x = x
+```
+```swift
+func id<A>(a: A) -> A {
```

Comment:
I would use `_` before the parameter name, in this way we don't need to specify it when using the function (it doesn't add value to the usage in my opinion.
```suggestion
func id<A>(_ a: A) -> A {
```